File: go/Euclid.go
package main

import (
	"math/rand"
	"time"
	"fmt"
	"math/big"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	a := rand.Intn(100)
	b := rand.Intn(100)
	GCD := Euclid(a, b)
	fmt.Println(a, b)
	fmt.Println(GCD)
	fmt.Println(GCD == (int)(gcd(uint64(a), uint64(b))))
}

func Euclid(a, b int) int {
	x, y := a, b
	if a < b {
		x, y = b, a
	}
	for {
		if x%y == 0 {
			return y
		} else {
			tmp := x % y
			x = y
			y = tmp
		}
	}
}

func gcd(m, n uint64) uint64 {
	x := new(big.Int)
	y := new(big.Int)
	z := new(big.Int)
	a := new(big.Int).SetUint64(m)
	b := new(big.Int).SetUint64(n)
	return z.GCD(x, y, a, b).Uint64()
}

File: java/src/Euclid.java
public class Euclid {
    int x, y;

    public void set(int a, int b) {
        this.x = a;
        this.y = b;
        if (a < b) {
            this.x = b;
            this.y = a;
        }
    }

    public int getGCD() {
        int tmp;
        while ((tmp = this.x % this.y) != 0) {
            this.x = this.y;
            this.y = tmp;
        }
        return y;
    }
}
